client/my-sites/upgrades/checkout/checkout.jsx

```diff
@@ -195,8 +199,13 @@ const Checkout = React.createClass( {
 			return selectedSiteSlug
 				? `/plans/${ selectedSiteSlug }/thank-you`
 				: '/checkout/thank-you/plans';
-		} else if ( cart.create_new_blog && cartItems.hasDomainRegistration( cart ) && ! cartItems.hasPlan( cart ) ) {
-			const domainName = cartItems.getDomainRegistrations( cart )[ 0 ].meta;
+		} else if ( cart.create_new_blog && ! cartItems.hasPlan( cart ) ) {
```

Comment:
Just copied the old code. If a user bought a plan, than it's not a `domain-only` purchase, and we can just switch back to the old flow.